src/hclib-locality-graph.c

```diff
@@ -1151,12 +1158,9 @@ hclib_locale_t *hclib_get_closest_locale_of_types(hclib_locale_t *locale,
                 to_visit[to_visit_index++] = i;
```

Comment:
Since to_visit_index starts from 1, to_visit[to_visit_index++] can go to out of bounds in case the the if condition is true on all iterations. If that case does not happen, then it is good to go.

Reply:
This should never be possible, because there are at most n_locales and the contains(i, to_visit, to_visit_index) validates that this list does not already contain this item. So even though we may evaluate the conditional around this stement more than n_locales times, we will only ever hit this conditional n_locales-1 times.

Reply:
Good to merge.